Yoast/wordpress-seo
# 12288 remove implementation of handleWorkerError

## Summary

This PR can be summarized in the following changelog entry:

* _[not-user-facing]_ Removes the handling of recalibration-specific errors in the analysis web worker.

## Relevant technical choices:

* Since only recalibration errors were handled, it leaves us with no error handling at all. 
  * To avoid having to reimplement functionality, we decided to leave the error catching code in, but remove the handling of the recalibration error. Leaving us with an empty error handling method (`handleWorkerError`).
* I removed and adapted the `handleWorkerError` tests as well to reflect the changes made in this PR. E.g. removed the testing of the recalibration error.

## Test instructions

This PR can be tested by following these steps:

* Open a post in WordPress.
* Add, change and delete some content. Everything should work as expected, with no errors.

To test if no error message warning box is shown on screen, we must create an error in the analysis web worker.
* Make sure that you have a local clone of `YoastSEO.js` project on your computer and that you have linked it to your local `wordpress-seo` repository using `yarn link yoastseo`.
* Open the `YoastSEO.js` project in your editor of choice (e.g. PHPStorm).
* Navigate in your editor to `YoastSEO.js/src/worker/AnalysisWebWorker.js`.
* The `handleMessage` handles all messages to the web worker, so is a prime candidate for injecting an error.
  * Add `throw new Error( "An error in the web worker!" );` somewhere in the `handleMessage` method to create an error.

Now for testing if no message is shown:
* Make sure that you have the Yoast development server running.
* Open a post in WordPress.
* The readability and SEO analysis should keep spinning.
* Open the developer console.
  * It should have an error message saying:
    `An error occurred while running the analysis.	Error: An error in the web worker!`

## UI changes

## Quality assurance


Fixes #12288
